### PR TITLE
feat(AppHeaderLink): link AppHeaderElement

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

Create AppHeaderLink component to be part of AppHeaderElements.

image

Part of the [Global Nav Unification](https://codecademy.atlassian.net/browse/EGG-642) project.

This PR comes after [EGG-539](https://codecademy.atlassian.net/browse/EGG-539) and [EGG-540](https://codecademy.atlassian.net/browse/EGG-540), work done with @saghdaey which resulted in a [draft PR combining all GlobalHeader work](https://github.com/Codecademy/client-modules/pull/1265). We are now breaking up this branch to merge the following pieces separately:

1. [EGG-643](https://codecademy.atlassian.net/browse/EGG-643): AppHeaderLogo (#1277)
2. [EGG-644](https://codecademy.atlassian.net/browse/EGG-644): AppHeaderLink (#1278)
3. [EGG-645](https://codecademy.atlassian.net/browse/EGG-645): AppHeaderDropdown (#1279)
4. [EGG-646](https://codecademy.atlassian.net/browse/EGG-646): AppHeader
5. [EGG-647](https://codecademy.atlassian.net/browse/EGG-647): GlobalHeader

This PR adds unit tests to AppHeaderLink.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-644](https://codecademy.atlassian.net/browse/EGG-644)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.394371.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60006a885b82a700a42a3530--gamut-preview.netlify.app>https://60006a885b82a700a42a3530--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60006a885b82a700a42a3530>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.a0f497.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60009af44184b03c00426483--gamut-preview.netlify.app>https://60009af44184b03c00426483--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60009af44184b03c00426483>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.f1aa4f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000a7311a76d84190f1bd70--gamut-preview.netlify.app>https://6000a7311a76d84190f1bd70--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000a7311a76d84190f1bd70>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.0ab427.0<br/>@codecademy/styleguide@21.10.1-alpha.0ab427.0<br/></pre>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+});
+
+const AppHeaderLinkButtonInner = styled(Box)`
+  white-space: nowrap;
+`;
+
+export type AppHeaderLinkProps = {
+  item: AppHeaderLinkItem;
+  // onClick: HeaderClickHandler;
+  onClick: (event: React.MouseEvent) => {}; // PLACEHOLDER until GlobalHeader is merged
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx:28]: Could we remove these comments? Or at the very least remove the first one and switch the other one to:
```
  onClick: (event: React.MouseEvent) => {}; // TODO switch to HeaderClickHandler 
```

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export const AppHeaderLink: React.FC<AppHeaderLinkProps> = ({
+  item,
+  onClick,
+}) => {
+  return (
+    <AppHeaderLinkButtonOuter
+      data-testid={item.dataTestId}
+      href={item.href}
+      // onClick={(event: React.MouseEvent) => onClick(event, item)}
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx:39]: Same, do we need this comment?

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx (hunk 1 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx:28]: Yeah I'm cool with that!

**Review by hmafzal — APPROVED** [review]
LGTM!

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx (hunk 1 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx:28]: I have a similar thing going on #1279 too - I'll do the same there

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.c35b6b.0<br/>@codecademy/styleguide@21.10.1-alpha.c35b6b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000add01a76d84389f1c028--gamut-preview.netlify.app>https://6000add01a76d84389f1c028--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000add01a76d84389f1c028>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000b0984184b0561242644a--gamut-preview.netlify.app>https://6000b0984184b0561242644a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000b0984184b0561242644a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.8ab0bb.0<br/>@codecademy/styleguide@21.10.1-alpha.8ab0bb.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.3836c1.0<br/>@codecademy/styleguide@21.11.1-alpha.3836c1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000bf791a76d85c36f1bf77--gamut-preview.netlify.app>https://6000bf791a76d85c36f1bf77--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000bf791a76d85c36f1bf77>Deploy Logs</a></p>

**Comment by saghdaey** [comment]
https://6000bf791a76d85c36f1bf77--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-appheaderlink--app-header-link WHO IS SHE!!!!! A PLATFORM ENG! 👏🏼✨👏🏼✨

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/__tests__/AppHeaderLink-test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+const renderAppHeaderLink = () => {
+  return render(
+    <ThemeProvider theme={theme}>
+      <AppHeaderLink {...props} />
+    </ThemeProvider>
+  );
+};
+
+describe('AppHeaderLink', () => {
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/__tests__/AppHeaderLink-test.tsx:31]: I'm not of the opinion that testing basic rendering or other non-logic flows (like, anything that happens no matter what) is super valuable for tests. I think the things you do want to test are events and logical branches. Testing the basics of the component feels to me like restating the component itself, which means these are more likely to only break when the component legitimately does need to change.

What do you think? Up for switching this to just testing that `onClick` happens?

**Review by JoshuaKGoldberg — APPROVED** [review]
A great start!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.5b0d6e.0<br/>@codecademy/styleguide@21.11.1-alpha.5b0d6e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600114caa1262e8369c70ac0--gamut-preview.netlify.app>https://600114caa1262e8369c70ac0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600114caa1262e8369c70ac0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.1772eb.0<br/>@codecademy/styleguide@21.11.1-alpha.1772eb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001163c75dde56c3d104eef--gamut-preview.netlify.app>https://6001163c75dde56c3d104eef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001163c75dde56c3d104eef>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/__tests__/AppHeaderLink-test.tsx (hunk 3 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/__tests__/AppHeaderLink-test.tsx:31]: Yes! Makes sense and I'm completely up for that - thanks Josh. Removed unnecessary tests here and #1279

**Pull request merged**